.github/workflows/pytest-workflow.yml

```diff
@@ -8,18 +8,27 @@ jobs:
     outputs:
       # Expose matched filters as job 'modules' output variable
       modules: ${{ steps.filter.outputs.changes }}
+      num_modules: ${{ steps.yq_modules.outputs.magic_number }}
     steps:
       - uses: actions/checkout@v2
       - uses: dorny/paths-filter@v2
         id: filter
         with:
           filters: 'tests/config/pytest_software.yml'
 
+      - uses: mikefarah/yq@v4.6.3
+        name: Install yq
+        id: yq_modules
+        with:
+          cmd: |
+            output=$(echo ${{ steps.filter.outputs.changes }} | yq e 'length' -)
+            echo "::set-output name=magic_number::$output"
+
   test:
     runs-on: ubuntu-20.04
     name: ${{ matrix.tags }} ${{ matrix.profile }} ${{ matrix.nxf_version }}
     needs: changes
-    if: needs.changes.outputs.modules != '[]'
+    if: needs.changes.outputs.modules != '[]' && needs.changes.outputs.num_modules <= 256
```

Comment:
This won't run, period if we have more than 256 tests?

Reply:
That's the idea! For my next trick I'm going to have a job that only runs if the changes are greater than 256, and it's just going to run everything for now.

Reply:
Cool.

Reply:
Nice :ok_hand: